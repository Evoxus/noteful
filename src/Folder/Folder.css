.Folder {
  display: block;
  text-align: center;
  margin: 7px 2px;
  padding: 10px;
  width: 95%;
  max-width: 100px;
  border: 1px solid white;
  text-decoration: none;
  color: white;
}
.Folder:hover {
  background-color: #2C4C4C;
}
.Folder.active {
  background-color: #406565;
}

.noteCount {
  font-size: x-small;
  color: #0C3030;
}