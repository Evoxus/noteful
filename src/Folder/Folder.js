import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { countNotesForFolder } from '../helperFunctions';
import './Folder.css';

export default function Folder(props){
  return(
    <NavLink className='Folder' to={`/folder/${props.folder.id}`}>
      {props.folder.folder_name}
      <span className='noteCount'>  {countNotesForFolder(props.notes, props.folder.id)}</span>
    </NavLink>
  )
}

Folder.defaultProps = {
  folder: {
    id: 1,
    name: 'Folder'
  }
}

Folder.propTypes = {
  folder: PropTypes.object.isRequired
}