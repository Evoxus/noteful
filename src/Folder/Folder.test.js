import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Folder from './Folder';

describe('App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter> <Folder /> </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('snapshot test', () => {
    const wrapper = shallow(<Folder />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})