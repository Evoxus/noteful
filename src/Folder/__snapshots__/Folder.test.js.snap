// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot test 1`] = `
<NavLink
  className="Folder"
  to="/folder/1"
>
  Folder
</NavLink>
`;
